contracts/SpecificGroupStrategy.sol

```diff
@@ -0,0 +1,643 @@
+        (groups, votes) = generateWithdrawalVoteDistributionTransfer(
+            group,
+            celoWithdrawalAmount,
+            stCeloWithdrawalAmount
+        );
+    }
+
+    /**
+     * @notice Generates groups and votes to distribute votes to.
+     * @param group The validator group that we want to deposit to or transfer from.
+     * @param celoAmount The amount of CELO.
+     * @param stCeloAmount The amount of stCELO.
+     * @return finalGroups The groups to withdraw from.
+     * @return finalVotes The amount to withdraw from each group.
+     */
+    function generateDepositVoteDistribution(
+        address group,
+        uint256 celoAmount,
+        uint256 stCeloAmount
+    ) external onlyManager returns (address[] memory finalGroups, uint256[] memory finalVotes) {
+        votedGroups.add(group);
+        updateGroupStCelo(group, stCeloAmount, true);
+
+        if (groupHealth.isGroupValid(group) && !blockedGroups.contains(group)) {
+            uint256 receivableVotes = IManager(manager).getReceivableVotesForGroup(group);
+            uint256 votesToBeScheduledForSpecificGroup = Math.min(receivableVotes, celoAmount);
+
+            celoAmount -= votesToBeScheduledForSpecificGroup;
+            if (celoAmount > 0) {
+                // overflow
+                (address[] memory groups, uint256[] memory votesForGroups) = defaultStrategy
+                    .generateDepositVoteDistribution(celoAmount, group);
+                updateOverflowGroup(group, IManager(manager).toStakedCelo(celoAmount), true);
+                finalGroups = new address[](groups.length + 1);
+                finalVotes = new uint256[](groups.length + 1);
+                for (uint256 i = 0; i < groups.length; i++) {
+                    finalGroups[i] = groups[i];
+                    finalVotes[i] = votesForGroups[i];
+                }
+                finalGroups[groups.length] = group;
+                finalVotes[groups.length] = votesToBeScheduledForSpecificGroup;
+            } else {
+                uint256 votesRemaining = account.getCeloForGroup(group);
+                if (votesRemaining < celoAmount) {
+                    revert GroupNotBalancedOrNotEnoughStCelo(group, celoAmount, votesRemaining);
+                }
+                finalGroups = new address[](1);
+                finalVotes = new uint256[](1);
+                finalGroups[0] = group;
+                finalVotes[0] = votesToBeScheduledForSpecificGroup;
+            }
+        } else {
+            (finalGroups, finalVotes) = defaultStrategy.generateDepositVoteDistribution(
+                celoAmount,
+                group
+            );
+            updateUnhealthyGroupStCelo(group, stCeloAmount, true);
+        }
+    }
+
+    /**
+     * @notice Used when validator gets unhealthy and we need to move funds to default strategy
+     * @param group The group address.
+     */
+    function rebalanceWhenHealthChanged(address group) external {
```

Comment:
When is this function called? will it be by a bot on a regular schedule?

Reply:
Yep, it will be called by bot for groups that are eligable ->

when group unhealthy and it has some stCELO that is not moved to DefaultStrategy
when group healthy and it has some stCELO that is still in DefaultStrategy